perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/MembersManagerBlImpl.java

```diff
@@ -2284,15 +2284,21 @@ public Member createSponsoredMember(PerunSession session, Vo vo, String namespac
 			Map<String, String> mapName = new HashMap<>();
 			mapName.put("guestName", name);
 			// create sponsored member
-			RichMember richMember;
+			User user;
 			try {
-				richMember = getRichMember(sess, createSponsoredMember(sess, vo, namespace, mapName, password, sponsor, asyncValidation));
+				user = perunBl.getUsersManagerBl().getUserByMember(sess, createSponsoredMember(sess, vo, namespace, mapName, password, sponsor, asyncValidation));
+				// get login to return
+				String login = perunBl.getAttributesManagerBl().getAttribute(sess, user, PasswordManagerModule.LOGIN_PREFIX + namespace).valueAsString();
+				Map<String, String> statusWithLogin = new HashMap<>();
+				statusWithLogin.put("status", "OK");
+				statusWithLogin.put("login", login);
+				statusWithLogin.put("password", password);
+				result.put(name, statusWithLogin);
 			} catch (Exception e) {
-				throw new InternalErrorException("These accounts had been created before exception was thrown: " + result, e);
+				Map<String, String> status = new HashMap<>();
+				status.put("status", e.getMessage());
```

Comment:
I would start this message like "ERROR: " + message.

Reply:
I think that every exception message implicitly starts with "Error <code>: <message>".